### PR TITLE
 Ref[Tracker][53139][Admin add showtimes for movies] 

### PR DESCRIPTION
**Pull request by congth2001**
## Related Tickets
- [#53139](https://edu-redmine.sun-asterisk.vn/issues/53139)

## WHAT (optional)
- Admin can add valid showtime for any movies

## HOW
- Validates date time for showtimes using custom method
- HTML, CSS, Bootstrap for showtimes list of movies and form to add showtimes
- Using nested resource showtimes for movies to routes showtime pages by movie id

## Evidence (Screenshot or Video)
Peek 2022-09-26 23-46

### CODE REVIEW

**Comment by congth2001** [comment]
ready

File: app/models/show_time.rb [hunk 1]
```diff
@@ -2,4 +2,28 @@ class ShowTime < ApplicationRecord
...
   belongs_to :room
+
+  validate :valid_format_showtime
+  validate :valid_overlap_showtime, if: :valid_format_showtime
+
+  def valid_overlap_showtime
+    return if ShowTime.where("room_id = ? AND
+                              ((start_time < ? AND end_time > ?)
+                                OR (end_time > ? AND start_time < ?)
+                                OR (start_time = ? AND end_time = ?))",
```
Comment (ghost) [inline app/models/show_time.rb]: Viết scope Em nhé và sử dụng query interface

File: app/views/admin/shared/_showtimes_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      <label class="col-sm-2 col-form-label"><%= t ".end_time" %></label>
+      <div class="col-sm-10">
+        <%= f.datetime_field :end_time %>
+      </div>
+    </div>
+    
+    <div class="form-group row">
+      <label class="col-sm-2 col-form-label"><%= t ".room" %></label>
+      <div class="col-sm-10">
+        <%= f.select :room_id, Room.all.map{|room|["#{room.name}, #{room.cinema.name}", room.id]},
```
Comment (ghost) [inline app/views/admin/shared/_showtimes_form.html.erb]: Áp dụng grouped_options_for_select

File: app/views/admin/show_times/_show_time.html.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<tr>
+  <td><%= @stt %></td><% @stt += 1 %>
+  <td><%= show_time.start_time.strftime(t "time.formats.hms") %></td>
```
Comment (ghost) [inline app/views/admin/show_times/_show_time.html.erb]: Dùng i18n time format

**Review by ghost — COMMENTED** [review]
cmt

**Comment by congth2001** [comment]
ready

File: app/views/admin/shared/_showtimes_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      <label class="col-sm-2 col-form-label"><%= t ".start_time" %></label>
+      <div class="col-sm-10">
+        <%= f.datetime_field :start_time %>
+      </div>
+    </div>
+
+    <div class="form-group row">
+      <label class="col-sm-2 col-form-label"><%= t ".room" %></label>
+      <div class="col-sm-10">
+        <% grouped_options =  Cinema.all.map{|cinema| [cinema.name, cinema.rooms.map{|room| [room.name, room.id]}]} %>
```
Comment (ghost) [inline app/views/admin/shared/_showtimes_form.html.erb]:
```suggestion
        <% grouped_options =  Cinema.all.map{|cinema| [cinema.name, cinema.rooms.pluck(:name, :id])]} %>
```

**Review by ghost — COMMENTED** [review]
cmt

**Comment by congth2001** [comment]
ready

File: app/views/admin/shared/_showtimes_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      <label class="col-sm-2 col-form-label"><%= t ".start_time" %></label>
+      <div class="col-sm-10">
+        <%= f.datetime_field :start_time %>
+      </div>
+    </div>
+
+    <div class="form-group row">
+      <label class="col-sm-2 col-form-label"><%= t ".room" %></label>
+      <div class="col-sm-10">
+        <% grouped_options =  Cinema.all.map{|cinema| [cinema.name, cinema.rooms.pluck(:name, :id)]} %>
```
Comment (anhnt-1044) [inline app/views/admin/shared/_showtimes_form.html.erb]: query ở view?

**Comment by congth2001** [comment]
ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**